# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MIR.h
# Commit: f1af63fb5986
# Full Hash: f1af63fb5986e30fa5695a26b9ed7cd506475e1d
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-09-09 16:10:02
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 6: Force Ion to preserve its tls. r=lth
#   
#   Let's see following code and let's assume that wasm-compiler is ion:
#   
#   call foo
# ==============================================================================

diff -r facc78377809 -r f1af63fb5986 js/src/jit/MIR.h
--- a/js/src/jit/MIR.h	Wed Sep 09 05:43:19 2020 +0000
+++ b/js/src/jit/MIR.h	Wed Sep 09 08:55:17 2020 +0000
@@ -13125,10 +13125,12 @@
   ABIArg abi() const { return abi_; }
 };
 
-class MWasmReturn : public MAryControlInstruction<1, 0>,
+class MWasmReturn : public MAryControlInstruction<2, 0>,
                     public NoTypePolicy::Data {
-  explicit MWasmReturn(MDefinition* ins) : MAryControlInstruction(classOpcode) {
+  MWasmReturn(MDefinition* ins, MDefinition* tls)
+      : MAryControlInstruction(classOpcode) {
     initOperand(0, ins);
+    initOperand(1, tls);
   }
 
  public:
@@ -13136,9 +13138,12 @@
   TRIVIAL_NEW_WRAPPERS
 };
 
-class MWasmReturnVoid : public MAryControlInstruction<0, 0>,
+class MWasmReturnVoid : public MAryControlInstruction<1, 0>,
                         public NoTypePolicy::Data {
-  MWasmReturnVoid() : MAryControlInstruction(classOpcode) {}
+  explicit MWasmReturnVoid(MDefinition* tls)
+      : MAryControlInstruction(classOpcode) {
+    initOperand(0, tls);
+  }
 
  public:
   INSTRUCTION_HEADER(WasmReturnVoid)